File: 2724-convert-an-array-into-a-2d-array-with-conditions/convert-an-array-into-a-2d-array-with-conditions.js
/**
 * @param {number[]} nums
 * @return {number[][]}
 */
var findMatrix = function(nums) {
    let ans = [];
    let map = new Map();
    let max = 0;
    for(let num of nums) {
        let curr = 1;
        if(!map.has(num)) map.set(num, curr);
        else {
            curr = map.get(num)+1;
            map.set(num, curr);
        }
        max = Math.max(max, curr);
    }
    for(let i=0; i<max; i++) {
        ans.push([]);
    }
    for(let [key, value] of map) {
        for(let i=0; i<value; i++) {
            ans[i].push(key);
        }
    }
    return ans;
};

File: 2724-convert-an-array-into-a-2d-array-with-conditions/convert-an-array-into-a-2d-array-with-conditions.java
class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        var ans = new ArrayList<List<Integer>>();
        var map = new HashMap<Integer, Integer>();
        for(int num : nums) {
            map.put(num, map.getOrDefault(num, 0) + 1);
        }
        while(!map.isEmpty()) {
            var list = new ArrayList<Integer>();
            var it = map.entrySet().iterator();
            while(it.hasNext()) {
                var entry = it.next();
                var key = entry.getKey();
                var value = entry.getValue();
                list.add(key);
                if(value == 1) {
                    it.remove();
                }
                else {
                    map.put(key, value-1);
                }
            }
            ans.add(list);
        }
        return ans;
    }
}